From 7bc3b341d90c95dc17d3da0930801006b00d2046 Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 4 Mar 2021 14:33:09 +0100
Subject: [PATCH] ARM: dts: bcm2711: yellow: Mux UART4 for SiLabs radio module

Enable UART4 by default and mux pins including hardware flow control.

Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 .../dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts   | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts b/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts
index f92936d7cf16b..fa93ab5ca639b 100644
--- a/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts
+++ b/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts
@@ -15,6 +15,10 @@ / {
 	compatible = "raspberrypi,4-compute-module-ha-yellow", "raspberrypi,4-compute-module", "brcm,bcm2711";
 	model = "Raspberry Pi Compute Module 4 on Home Assistant Yellow";
 
+	aliases {
+		serial4 = &uart4;
+	};
+
 	chosen {
 		/* 8250 auxiliary UART instead of pl011 */
 		stdout-path = "serial1:115200n8";
@@ -243,6 +247,14 @@ &uart1 {
 	status = "okay";
 };
 
+/* uart4 for Zigbee */
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart4_pins>;
+	uart-has-rtscts;
+	status = "okay";
+};
+
 &vc4 {
 	status = "okay";
 };
@@ -467,9 +479,9 @@ uart3_pins: uart3_pins {
 	};
 
 	uart4_pins: uart4_pins {
-		brcm,pins = <8 9>;
+		brcm,pins = <8 9 10 11>;
 		brcm,function = <BCM2835_FSEL_ALT4>;
-		brcm,pull = <0 2>;
+		brcm,pull = <0 2 2 0>;
 	};
 
 	uart5_pins: uart5_pins {
